src/System.Slices/src/System/MemoryUtils.Other.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    internal static partial class MemoryUtils
+    {
+        /// <summary>
+        /// platform independent fast memory comparison based on idea from String.EqualsHelper 
+        /// </summary>
+        internal static bool MemCmp<[Primitive]T>(Span<T> first, Span<T> second)
+            where T : struct
+        {
+            unsafe
+            {
+                // prevent GC from moving memory
+                var firstPinnedHandle = GCHandle.Alloc(first.Object, GCHandleType.Pinned);
+                var secondPinnedHandle = GCHandle.Alloc(second.Object, GCHandleType.Pinned);
+
+                try
+                {
+                    byte* firstPointer = (byte*)PtrUtils.ComputeAddress(first.Object, first.Offset).ToPointer();
+                    byte* secondPointer = (byte*)PtrUtils.ComputeAddress(second.Object, second.Offset).ToPointer();
+
+                    long bytesCount = first.Length * PtrUtils.SizeOf<T>();
+
+                    while (bytesCount >= 20)
+                    {
+                        if (*(int*)firstPointer != *(int*)secondPointer) break;
+                        if (*(int*)(firstPointer + 4) != *(int*)(secondPointer + 4)) break;
+                        if (*(int*)(firstPointer + 8) != *(int*)(secondPointer + 8)) break;
```

Comment:
Could it be advantageous to use void* here instead of int. Those also can be compared, but could be faster on 64bit.
```C#
            if (*(void**)firstPointer != *(void**)secondPointer) break;
```
you will need to scale the stepping by IntPtr.Size